src/ProductDetail/ProductDetail.js

```diff
@@ -0,0 +1,151 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import { useData } from "../Products";
+import { useUser } from "../User";
+import { Link } from "react-router-dom";
+import { found, isProdInCart, isProdInWishList, loginAlert } from "../utils";
+import {
+  addToCartApi,
+  addToWishListApi,
+  deleteFromWishListApi,
+} from "../api-calls";
+import { useAuth } from "../Auth";
+import { LoginAlertModal } from "../LoginAlert";
+import Loading from "./../images/loading.svg";
+import "./ProductDetail.css";
+
+export const ProductDetail = () => {
+  const { id } = useParams();
+  const { status, productDetail, dispatch } = useData();
+  const { userState, userDispatch } = useUser();
+  const { token } = useAuth();
+  const [showModal, setShowModal] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        dispatch({
+          type: "STATUS",
+          payload: { loading: "Loading data from server..." },
+        });
+        const response = await axios.get(
+          `https://api-prestore.prerananawar1.repl.co/products/${id}`
+        );
+        const data = response.data.product;
+        dispatch({ type: "PRODUCT_DETAIL", payload: data });
+        dispatch({
```

Comment:
If you are setting the product details in context, it's better to clear the context in the cleanup function, so that when the user goes to any other product wrong data is not shown

Reply:
Thank you for this one was really helpful